redstone_serde/deserialize/play.ts

```diff
@@ -1,97 +1,607 @@
+  };
+}
+
+function messageAcknowledgment(reader: Reader): MessageAcknowledgementPayload {
+  return {
+    messageCount: reader.getVarInt(),
+  };
+}
+
+function chatCommand(reader: Reader): ChatCommandPayload {
+  return {
+    command: reader.getString(),
+    timestamp: reader.getBigInt64(),
+    salt: reader.getBigInt64(),
+    argumentSignatures: new Array(reader.getVarInt()).fill(undefined).map(() => ({
+      argumentName: reader.getString(),
+      signature: reader.getSlice(256),
+    })),
+    messageCount: reader.getVarInt(),
+    acknowledged: reader.getSlice(3),
+  };
+}
+
+function chatMessage(reader: Reader): ChatMessagePayload {
+  return {
+    message: reader.getString(),
+    timestamp: reader.getBigInt64(),
+    salt: reader.getBigInt64(),
+    signature: reader.getInt8() ? reader.getSlice(256) : undefined,
+    messageCount: reader.getVarInt(),
+    acknowledged: reader.getSlice(3),
+  };
+}
+
+function playerSession(reader: Reader): PlayerSessionPayload {
+  return {
+    sessionID: deserializeUUID(reader),
+    publicKey: {
+      expiresAt: reader.getBigInt64(),
+      publicKey: reader.getSlice(reader.getVarInt()),
+      keySignature: reader.getSlice(reader.getVarInt()),
+    },
+  };
+}
+
+function clientCommand(reader: Reader): ClientCommandPayload {
+  return {
+    actionID: reader.getVarInt(),
+  };
+}
+
+function clientInformation(reader: Reader): ClientInformationPayload {
+  const getDisplayedSkinParts = (skinFlagsByte: number) => ({
+    cape: !!(skinFlagsByte & 0x01),
+    jacket: !!(skinFlagsByte & 0x02),
+    leftSleeve: !!(skinFlagsByte & 0x04),
+    rightSleeve: !!(skinFlagsByte & 0x08),
+    leftPantsLeg: !!(skinFlagsByte & 0x10),
+    rightPantsLeg: !!(skinFlagsByte & 0x20),
+    hat: !!(skinFlagsByte & 0x40),
+  });
+
+  return {
+    locale: reader.getString(),
+    viewDistance: reader.getInt8(),
+    chatMode: reader.getVarInt(),
+    chatColors: !!reader.getInt8(),
+    displayedSkinParts: getDisplayedSkinParts(reader.getUint8()),
+    mainHand: reader.getVarInt(),
+    enableTextFiltering: !!reader.getInt8(),
+    allowServerListings: !!reader.getInt8(),
+  };
+}
+
+function commandSuggestionsRequest(reader: Reader): CommandSuggestionsRequestPayload {
+  return {
+    transactionID: reader.getVarInt(),
+    text: reader.getString(),
+  };
+}
+
+function clickContainerButton(reader: Reader): ClickContainerButtonPayload {
+  return {
+    windowID: reader.getInt8(),
+    buttonID: reader.getInt8(),
+  };
+}
+
+function clickContainer(reader: Reader): ClickContainerPayload {
+  return {
+    windowID: reader.getUint8(),
+    stateID: reader.getVarInt(),
+    slot: reader.getInt16(),
+    button: reader.getInt8(),
+    mode: reader.getVarInt() as 0 | 1 | 2 | 3 | 4 | 5 | 6,
+    slots: new Array(reader.getVarInt()).fill(undefined).map(() => ({
+      position: reader.getInt16(),
+      slot: readSlot(reader),
+    })).sort((a, b) => a.position - b.position).map(({ slot }) => slot),
```

Comment:
Does this need to be sorted?

Reply:
As discussed on discord: Does not have to be sorted and will be rewritten a little in a follow up PR